### PR TITLE
Contexts (from master)

### PR DESCRIPTION
**Pull request by tmbb**
Created new contexts branch from master.
Had to add "phoenix-1.3.0-rc.2" as a dependency to be able to reuse some of the generators.
It's very import to use the latest RC (or at leas not the first) because of the naming of the tables that was changed.
In previous RCs, the table names were prefixed by the contexts, and were later removed due to user feedback (rightly, in my opinion) in later versions.

### CODE REVIEW

**Comment by tmbb** [comment]
Thing are working bit thi PR doesn't address the problem described here: https://github.com/smpallen99/coherence/issues/229

File: README.md [hunk 1]
```diff
@@ -752,6 +752,63 @@ Coherence is a user management and authentication solution. Support for authoriz
 
 For an example of using [Canary](https://github.com/cpjk/canary) with Coherence, please visit the [CoherenceDemo canary branch](https://github.com/smpallen99/coherence_demo/tree/canary).
 
+## Contexts
+
+Phoenix 1.3 promotes contexts as an abstraction on top of Ecto schemas or models.
```
Comment (smpallen99) [inline README.md]: I don't think we need to explain Phoenix contexts here. Simply explain that a new context can be generated for the Coherence user schema.

File: lib/mix/tasks/coh.context.ex [hunk 2]
```diff
@@ -0,0 +1,187 @@
+defmodule Mix.Tasks.Coh.Context do
```
Comment (smpallen99) [inline lib/mix/tasks/coh.context.ex]: The task should be renamed to `coh.gen.context`

File: lib/mix/tasks/coh.context.ex [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+  @shortdoc "Generates a context with functions around an Ecto schema representing a User"
+
+  @moduledoc """
+  Generates a context with functions around an Ecto schema representing a User.
+      mix coh.context Accounts name:string email:string
+  """
```
Comment (smpallen99) [inline lib/mix/tasks/coh.context.ex]: Change this section to `## Examples`

Add an example to specify the schema name.
`coh.gen.schema Accounts name:string email:string --schema=Account`

File: lib/mix/tasks/coh.context.ex [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  alias Mix.Phoenix.{Context}
+  alias Mix.Tasks.Phx.Gen
+
+  @switches [binary_id: :boolean, table: :string, web: :string,
+             schema: :boolean, context: :boolean, context_app: :string]
+
+  @default_opts [schema: true, context: true]
+
+  # Hardcoded schema name:
+  @schema_name "User"
```
Comment (smpallen99) [inline lib/mix/tasks/coh.context.ex]: The default schema name (if the --schema option is not provided) should be the last name of `Module.split Config.user_schema`

File: lib/mix/tasks/coh.context.ex [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+  @plural "users"
```
Comment (smpallen99) [inline lib/mix/tasks/coh.context.ex]: The table name should be retrieved from `Config.user_schema.__schema__(:source)`

File: lib/mix/tasks/coh.context.ex [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  """
+  
+  # Code shamelessly stolen from the Phx default context generator with minor changes
+
+  use Mix.Task
+
+  alias Mix.Phoenix.{Context}
+  alias Mix.Tasks.Phx.Gen
+
+  @switches [binary_id: :boolean, table: :string, web: :string,
```
Comment (smpallen99) [inline lib/mix/tasks/coh.context.ex]: `binary_id` option is not needed. You would determine if `:binary_id` should be used by running:

```
iex(12)> Coherence.Config.user_schema.__schema__(:type, Coherence.Config.user_schema.__schema__(:primary_key) |> hd)
:binary_id
```

File: README.md (hunk 1 above)
Comment (tmbb) [inline README.md]: Ok, I'll remove it

File: lib/mix/tasks/coh.context.ex (hunk 6 above)
Comment (tmbb) [inline lib/mix/tasks/coh.context.ex]: Cool, by introspecting the user model I can have some validation and save the user some typing.

**Comment by tmbb** [comment]
I'll work on this as soon as possible. Thanks for the feedback

**Comment by smpallen99** [comment]
This is very old and would take a lot of work to merge now. I'm closing this PR.

**Pull request closed without merging**